packages/rum/src/viewCollection.ts

```diff
@@ -165,8 +170,20 @@ function trackHistory(onHistoryChange: () => void) {
   window.addEventListener(DOM_EVENT.POP_STATE, monitor(onHistoryChange))
 }
 
-function areDifferentViews(previous: Location, current: Location) {
-  return previous.pathname !== current.pathname
+function trackHash(onHashChange: () => void) {
+  window.addEventListener('hashchange', monitor(onHashChange))
+}
+
+function areDifferentViews(previous: Location, current: Location): boolean {
+  return (
+    previous.pathname !== current.pathname ||
+    // the views are different when:
+    //  switching from a defined hash to no hash ('')
+    //  switching from a defined hash to a different and defined hash
+    //  switching from no hash ('') to a defined hash
+    (previous.hash && previous.hash !== current.hash) ||
+    Boolean(!previous.hash && current.hash)
```

Comment:
```suggestion
   previous.hash !== current.hash
```
if it is more than that, tests are missing

Reply:
ok we can simplify as if `previous` is not defined, and `current.hash= ''` (we don't have an hash change) but we have a pathname change and the previous `previous.pathname !== current.pathname` would trigger a new view anyway

Reply:
How previous can not be defined?